src/Admin/Views/Organizations/Edit.cshtml

```diff
@@ -117,228 +137,258 @@
-                <label asp-for="ExpirationDate"></label>
-                <input type="datetime-local" class="form-control" asp-for="ExpirationDate">
+            <div class="col-sm">
+                <div class="form-group">
+                    <label asp-for="ExpirationDate"></label>
+                    <input type="datetime-local" class="form-control" asp-for="ExpirationDate" readonly='@(!canEditLicensing)'>
+                </div>
             </div>
         </div>
-    </div>
-    <h2>Billing</h2>
-    <div class="row">
-        <div class="col-sm">
-            <div class="form-group">
-                <label asp-for="BillingEmail"></label>
-                <input type="email" class="form-control" asp-for="BillingEmail">
+    }
+
+    @if (canViewBilling)
+    {
+        <h2>Billing</h2>
+        <div class="row">
+            <div class="col-sm">
+                <div class="form-group">
+                    <label asp-for="BillingEmail"></label>
+                    <input type="email" class="form-control" asp-for="BillingEmail" readonly='@(!canEditBilling)'>
+                </div>
             </div>
-        </div>
-        <div class="col-sm">
-            <div class="form-group">
+            <div class="col-sm">
                 <div class="form-group">
-                    <label asp-for="Gateway"></label>
-                    <select class="form-control" asp-for="Gateway"
-                            asp-items="Html.GetEnumSelectList<Bit.Core.Enums.GatewayType>()">
-                        <option value="">--</option>
-                    </select>
+                    <div class="form-group">
+                        <label asp-for="Gateway"></label>
+                        <select class="form-control" asp-for="Gateway" disabled='@(canEditBilling ? null : "disabled")'
+                                asp-items="Html.GetEnumSelectList<Bit.Core.Enums.GatewayType>()">
+                            <option value="">--</option>
+                        </select>
+                    </div>
                 </div>
             </div>
         </div>
-    </div>
-    <div class="row">
-        <div class="col-sm">
-            <div class="form-group">
-                <label asp-for="GatewayCustomerId"></label>
-                <div class="input-group">
-                    <input type="text" class="form-control" asp-for="GatewayCustomerId">
-                    <div class="input-group-append">
-                        <button class="btn btn-secondary" type="button" id="gateway-customer-link">
-                            <i class="fa fa-external-link"></i>
-                        </button>
+        <div class="row">
+            <div class="col-sm">
+                <div class="form-group">
+                    <label asp-for="GatewayCustomerId"></label>
+                    <div class="input-group">
+                        <input type="text" class="form-control" asp-for="GatewayCustomerId" readonly='@(!canEditBilling)'>
```

Comment:
❓ of curiosity: you use readonly for this input, but disabled for the gateway select on line 330. Is that intentional?

Reply:
Great question. `readonly` is preferable in a lot of ways - the conditional syntax is more readable, and the values of `readonly` elements bind to the model on save, unlike `disabled` values. But unfortunately `readonly` can’t be applied to checkboxes or radio buttons, so `disabled` must be used instead.

Reply:
@dgoodman-bw, ah that makes sense. Thank you!